compiler/rustc_typeck/src/check/demand.rs

```diff
@@ -355,6 +356,16 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 })
                 .filter_map(|variant| {
                     let sole_field = &variant.fields[0];
+
+                    if !sole_field.did.is_local()
+                        && !sole_field.vis.is_accessible_from(
+                            self.tcx.parent_module(expr.hir_id).to_def_id(),
+                            self.tcx,
+                        )
```

Comment:
```suggestion
                    if !sole_field.vis.is_accessible_from(
                            self.body_id.owner.to_def_id(),
                            self.tcx,
                        )
```

Reply:
No need to restrict to only foreign def id, and we can get a more accurate def id from `self.body_id.owner.to_def_id()` i think.

Reply:
Restriction to foreign def ids is deliberate, see the test: https://github.com/rust-lang/rust/blob/da2752e00f0139fb13282b2bd97aa0f8c665aee9/src/test/ui/mismatched_types/wrap-suggestion-privacy.rs#L10-L11

Reply:
Specifically " its privacy can be easily changed " as the justification would be nice to note

Reply:
Not sure how to present it in a good way, but I've tried 5bd88dfa8a0b5d5bcdd84d21e2f465719ce8c328

Reply:
haha I meant a comment in the source code, but this works too!

Reply:
lmao

it's bed time